BaseCollector.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+from os import getenv
+from modules.NetboxHelper import NetboxHelper
+from modules.VcenterConnection import VcenterConnection
+
+import logging
+import modules.TimedBlacklist as blacklist
+
+
+logger = logging.getLogger('esxi-exporter')
+
+
+class BaseCollector(ABC):
+
+    def __init__(self) -> None:
+        self.vcenter = VcenterConnection(getenv('VCENTER_URL'),
+                                         getenv('VCENTER_USER'),
+                                         getenv('VCENTER_PASSWORD'))

```

Comment:
This shows a basic problem the exporter has. This indicates that you can only start the exporter running against one VCenter basically. The following should be the case:

1. read the atlas.yaml (mounted in the container)
2. get all vcenters from this file
3. have the collector running against all vcenters in that region, not just one.

Atlas can also give you the hosts, so no need to talk to netbox directly. AFAIK it should also include, if the hosts are active or not. You need to check this.